src/lib/remark-plugin-adjust-link-urls.ts

```diff
@@ -1,15 +1,57 @@
+import path from 'path'
 import { visit } from 'unist-util-visit'
 import { Link } from 'mdast'
 import { Plugin, Transformer } from 'unified'
 
+const getIsInternalPath = (url: string) => {
+	try {
+		new URL(url)
+		return false
+	} catch (e) {
+		return true
+	}
+}
+
 const remarkPluginAdjustLinkUrls: Plugin = ({
+	currentPath = '',
 	urlAdjustFn,
 }: {
+	currentPath: string
 	urlAdjustFn: (url: string) => string
 }): Transformer => {
 	return function transformer(tree) {
 		visit(tree, ['link', 'definition'], (node: Link) => {
-			node.url = urlAdjustFn(node.url)
+			let urlToAdjust = node.url
+
+			/**
+			 * This is to handle links like "variables/input", which are effectively
+			 * the same thing as "./input" in the context of its current page.
+			 */
+			const isInternalPath = getIsInternalPath(node.url)
+			const needsPreAdjustment = isInternalPath && !/^[./?#]/.test(node.url)
+			if (needsPreAdjustment) {
+				const currentPathParts = currentPath.split('/')
+				const urlToChangeParts = node.url.split('/')
+
+				const newPathPrefixParts = []
+				currentPathParts.find((currentPathPart) => {
```

Comment:
💭  I'm finding the use of `find` here as a way to construct `newPathPrefixParts` until we hit an overlapping path segment threw me off a bit. I wonder if a `for` loop through `currentPathParts` with a `break` once `matchesUrlToChangeFirstPart` is found might be more clear?

Reply:
💭 Had a related thought here - it seems like it's possible there might be weird edge cases where `currentPath` contains repeated parts - for example, https://developer.hashicorp.com/consul/docs/api-gateway/usage/usage. Maybe this relates to the broader idea that links like `variables/input` might be worth trying to filter out with content conformance, rather than correct at the this stage.